src/main/java/com/example/bookstore/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package com.example.bookstore.security;
+
+import com.example.bookstore.exception.LoginException;
+import com.example.bookstore.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username)
+                .orElseThrow(() -> new LoginException("Can't find user by email"));
```

Comment:
```suggestion
                .orElseThrow(() -> new LoginException("Can't find user by email"));
```
You can use UsernameNotFoundException